packages/e2e-tests/specs/editor/blocks/navigation.test.js

```diff
@@ -766,4 +770,58 @@ describe( 'Navigation', () => {
 			expect( await page.$x( NAV_ENTITY_SELECTOR ) ).toHaveLength( 1 );
 		} );
 	} );
+
+	describe( 'Permission based restrictions', () => {
+		it( 'shows a warning if user does not have permission to edit / update Navigations', async () => {
+			await createNewPost();
+			await insertBlock( 'Navigation' );
+
+			const startEmptyButton = await page.waitForXPath(
+				START_EMPTY_XPATH
+			);
+
+			// This creates an empty Navigation post type entity.
+			await startEmptyButton.click();
+
+			// Publishing the Post ensures the Navigation entity is saved.
+			// The Post itself is irrelevant.
+			await publishPost();
+
+			const username = 'contributoruser';
+
+			const contribUserPassword = await createUser( username, {
+				role: 'contributor',
+			} );
+
+			// Switch to a Contributor role user - they should not have
+			// permission to update Navigations.
+			await loginUser( username, contribUserPassword );
+
+			await createNewPost();
+
+			await insertBlock( 'Navigation' );
+
+			// Select the Navigation post created by the Admin early
+			// in the test.
+			const navigationPostCreatedByAdminName = 'Navigation';
+			const dropdown = await page.waitForXPath( SELECT_MENU_XPATH );
+			await dropdown.click();
+			const theOption = await page.waitForXPath(
+				`//*[contains(@class, 'components-menu-item__item')][ text()="${ navigationPostCreatedByAdminName }" ]`
+			);
+			await theOption.click();
+
+			// Make sure the snackbar error shows up
+			await page.waitForXPath(
+				`//*[contains(@class, 'components-snackbar__content')][ text()="You do not have permission to edit this Navigation. Any edits made will not be saved." ]`
+			);
+
+			// Tidy up after ourselves.
+			await deleteUser( username );
+
+			// Expect a console 403 for request to Navigation Areas.
+			// Todo: removed once Nav Areas are removed from the Gutenberg Plugin.
+			expect( console ).toHaveErrored();
```

Comment:
I thought this had been fixed already?

Reply:
Yes they are supposed to work in the Plugin, 

> ...so not sure why there would be a 403...

As with many cases, this user doesn't have _permissions_ to read them. Requires `edit_theme_options` which is effectively only `Admin` roles.

Notice how the error messages match up

https://github.com/WordPress/gutenberg/blob/4cedaf0c4abb0aeac4bfd4289d63e9889efe9733/lib/class-wp-rest-block-navigation-areas-controller.php#L81-L91
<img width="1275" alt="Screen Shot 2021-12-15 at 10 33 52" src="https://user-images.githubusercontent.com/444434/146170695-794ef8db-522f-4a2d-b398-00e6397f49a6.png">

> ....why it isn't encountered in other e2e tests.

That's because no other tests switch to a lower permission users 😞 

I'll add a comment to the test to explain.

Reply:
Added in https://github.com/WordPress/gutenberg/pull/37286/commits/cfc7301e32281ad6bd3c3364fac0fabea5e3dbb1

Reply:
Right, yep, that makes sense now, a comment would be very helpful, thanks!